analyzers/tests/SonarAnalyzer.UnitTest/TestCases/UriShouldNotBeHardcoded.cs

```diff
@@ -40,6 +40,7 @@ void InvalidCases(string s1, string s2)
             new Uri(new Uri("../stuff"), ("C:/test.txt")); // Noncompliant
             File.OpenRead(@"\\drive\foo.csv"); // Noncompliant
 
+            // The rule is not raising for non-english variable names:
```

Comment:
The case here is not that we don't check non-English names exactly -  but that when the variable name is split based on the cameCase ([see related lines in code](https://github.com/SonarSource/sonar-dotnet/blob/master/analyzers/src/SonarAnalyzer.Common/Rules/UriShouldNotBeHardcodedBase.cs#L142)) - none of the string in the result exists [in the defined set](https://github.com/SonarSource/sonar-dotnet/blob/master/analyzers/src/SonarAnalyzer.Common/Rules/UriShouldNotBeHardcodedBase.cs#L44).

Maybe you can change the comment and say that none of the words `web` or `chemin` are part of the predefined set of string used to detect potential uri variable names.

Reply:
After discussing offline with @cristian-ambrosini-sonarsource - we still want to mention that we don't support non-English variable names but also add that practically this is also happening due to the way the variables are being picked up by the rule.